EventListener/IsGrantedListener.php

```diff
@@ -72,6 +75,23 @@ public function onKernelController(FilterControllerEvent $event)
         }
     }
 
+    private function getArguments(FilterControllerArgumentsEvent $event)
+    {
+        $namedArguments = $event->getRequest()->attributes->all();
+        $argumentMetadatas = $this->argumentMetadataFactory->createArgumentMetadata($event->getController());
+
+        // loop over each argument value and its name from the metadata
+        foreach ($event->getArguments() as $index => $argument) {
+            if (!isset($argumentMetadatas[$index])) {
+                throw new \LogicException(sprintf('Could not find any argument metadata for argument %d of the controller.', $index));
+            }
+
+            $namedArguments[$argumentMetadatas[$index]->getName()] = $argument;
+        }
+
+        return $namedArguments;
+    }
```

Comment:
This is a technical "bummer". We can use `$event->getArguments()` to get the array of arguments to the controller, but we don't know what *name* of each argument. So, we use the `ArgumentMetadataFactory` to reverse-engineer the argument names. I can't think of a situation where this would *not* work, but it's a bit weird.